p2p/ipld/sample.go

```diff
@@ -0,0 +1,99 @@
+package ipld
+
+import (
+	"github.com/ipfs/go-cid"
+	"github.com/lazyledger/nmt/namespace"
+
+	"github.com/lazyledger/lazyledger-core/libs/rand"
+	"github.com/lazyledger/lazyledger-core/p2p/ipld/plugin/nodes"
+	"github.com/lazyledger/lazyledger-core/types"
+)
+
+// Sample is a point in 2D space over square.
+type Sample struct {
+	Row, Col uint32
+}
+
+// SampleSquare randomly picks *num* unique points from arbitrary *width* square
+// and returns them as samples.
+func SampleSquare(squareWidth uint32, num int) []*Sample {
+	ss := newSquareSampler(squareWidth, num)
+	ss.sample(num)
+	return ss.sampled()
+}
+
+// Leaf returns leaf info needed for retrieval using data provided with DAHeader.
+func (s *Sample) Leaf(dah *types.DataAvailabilityHeader) (cid.Cid, uint32, error) {
```

Comment:
nit: I'm not sure we need to use a pointer for `*Sample`

Reply:
Yeah, not using pointer would be better